tests/iana_to_rflx/generated_types/test_duplicate_identifiers

```diff
@@ -0,0 +1,9 @@
+   type Duplicate_Identifiers is
+      (
+       Duplicate_Identifier => 1,
+
+       Duplicate_Identifier_2 => 2,
+
+       Duplicate_Identifier_3 => 3)
+   with Size => 2, Always_Valid;
```

Comment:
Is that really necessary? This would require major changes to the existing code (a complete redesign in fact). 
Currently, if a record with name "A" is contained in one registry, its name is added to the `duplicates` list since it has not occurred earlier. The registry is immediately written after the last record in it has been processed. 
If another record with name "A" is contained in another registry later on, it is checked if the name is contained in the `duplicates` list, which is true. Then the name for the second record is modified, so that is unique (append the value to the name) and also written to the duplicates list and the output file.

To achieve what you suggested, all types must be written at once after the last registry has been processed, so that all record names can be modified at any point before writing the output.

Reply:
You are developing a long term solution here, thus we shouldn't discuss whether something is really necessary, but rather what is the best solution. From a user perspective, the current approach seems unintuitive to me. I also think that refactoring the code into a processing and generating phase could actually be an improvement.

Reply:
Done that in efbfa5f7. The result is visible e.g. in 5d4de235.